### PR TITLE
hotfix(condo): add retrying for deliverMessage task

### PR DESCRIPTION
**Pull request by VKislov**
Before release, you need to delete the task with the old config in production. So that after the release there is no duplication

### CODE REVIEW

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 1]
```diff
@@ -127,7 +129,14 @@ class AppleMessaging {
 
             logger.info({ msg: 'sendPush before request', headers, options, payload })
 
-            const stream = this.#session.request(headers)
+            let stream
+            for (let retryCounter = 0; retryCounter < 5; retryCounter++) {
```
Comment (AleX83Xpert) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:133]: Looks like this code will send 5 push messages.
Every loop iteration you rewrite the `stream` variable. After the loop you have another "onResponse" callback. Which stream this callback is?

You need to wait the answer for the first request. If the answer is erroneous you should repeat the request.

**Review by AleX83Xpert — CHANGES_REQUESTED** [review]

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 2]
```diff
@@ -127,11 +129,20 @@ class AppleMessaging {
...
+                stream = this.#session.request(headers)
```
Comment (pahaz) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:134]: Why not to use await ?

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 3]
```diff
@@ -127,11 +129,20 @@ class AppleMessaging {
...
+                stream.on('error', (error) => {
+                    logger.error({ msg: 'sendPush errored', failCounter: retryCounter + 1, headers, options, payload, streamError: error })
```
Comment (pahaz) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:136]: use `err` key for error! we need to save tranceback

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 4]
```diff
@@ -161,7 +161,12 @@ class AppleMessaging {
                 payload.aps.alert = { ...notification }
             }
 
-            const response = await this.sendPush(token, payload, options)
+            let response
+            for (let retryCounter = 0; retryCounter < 5; retryCounter++) {
```
Comment (AleX83Xpert) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:165]: It's better to move `5` to some constant

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 5]
```diff
@@ -161,7 +161,12 @@ class AppleMessaging {
...
+                response = await this.sendPush(token, payload, options)
+                if (response instanceof Error) continue
```
Comment (AleX83Xpert) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:167]: Maybe add some logs?
```suggestion
                if (response instanceof Error) {
                    logger.warn({ msg: '...' })
                    continue
                }
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/notification/adapters/apple/AppleMessaging.js [hunk 6]
```diff
@@ -130,9 +131,9 @@ class AppleMessaging {
             const stream = this.#session.request(headers)
 
             stream.on('response', this.getResponseHandler(stream, resolve, reject))
-            stream.on('error', (error) => {
-                logger.error({ msg: 'sendPush errored', headers, options, payload, streamError: error })
-                return resolve(error)
+            stream.on('error', (err) => {
+                logger.error({ msg: 'sendPush errored', headers, options, payload, err })
+                return resolve(err)
```
Comment (pahaz) [inline apps/condo/domains/notification/adapters/apple/AppleMessaging.js:136]: It looks so strange! you should use reject for such things

**Pull request merged**